nomos-services/chain/chain-leader/src/leadership.rs

```diff
@@ -124,4 +101,139 @@ impl Leader {
 
         None
     }
+
+    fn private_inputs_for_winning_utxo_and_slot(
+        &self,
+        utxo: &Utxo,
+        epoch: Epoch,
+        public_inputs: LeaderPublic,
+    ) -> LeaderPrivate {
+        // TODO: Get the actual witness paths and leader key
+        let aged_path = Vec::new(); // Placeholder for aged path
```

Comment:
This stuff should be retrieved from wallet right?

Reply:
Yeah I mean this PR was intended to mirror the current behaviour to pre-compute stuff for Blend. I did not want to change too much the current behaviour. I am addressing some of @davidrusu's comments, but some require bigger changes that should be tackled in a different PR. Also, I was not sure David or Giacomo were already working on them.

Reply:
aged_path and latest_path actually comes from the ledger (It's the path in the UtxoTree).

data flow is
1. fetch eligible notes from wallet
2. derive the {latest|aged|_paths from latest ledger utxo tree and epoch_state's utxo tree
3. prove PoL with that data

we'll fill this in soon